app/profiles/landlord/page.tsx

```diff
@@ -0,0 +1,91 @@
+import { createClient } from "@/utils/supabase/server"
+import { cookies } from "next/headers"
+import { NextPage } from "next"
+
+const landlordProfilePage: NextPage = async ({
+    searchParams
+}: {
+    searchParams?: {
+        landlordId?: string,
+    }
+}) => {
+    // check if a landlord id was provided
+    const landlordId = searchParams?.landlordId
+    if (!landlordId) return (
+        <div>
+            <h1>ERROR: No Landlord Id provided</h1>
+        </div>
+    )
+
+    // set up the supabase client
+    const cookieStore = cookies()
+    const supabase = createClient(cookieStore)
+
+    // check if a landlord with the provided id exists and get their info
+    const { data: landlordData, error: landlordError } = await supabase
+        .from('user_profiles')
+        .select('*')
+        .eq('user_id', landlordId)
+        .single()
+
+    if (landlordError || !landlordData) return (
+        <div>
+            <h1>ERROR: Landlord profile not found</h1>
+        </div>
+    )
```

Comment:
you might want to consider using [not-found](https://nextjs.org/docs/app/api-reference/functions/not-found) here